### DISCUSSION TITLE
Language.from_disk() does not load pipeline

### DISCUSSION THREAD

**Discussion by hertelm** [explosion/spaCy#5670] [Help: Coding & Implementations] [answered]
Loading a saved model via _from_disk(...)_ does not load the pipeline, but the example from the [docs](https://spacy.io/api/language#from_disk) suggests that I do not need to take care of this.

## How to reproduce the behaviour
Minimum example:
```
import spacy
from spacy.lang.en import English


model = spacy.load("en_core_web_sm")
print("Original pipeline:", model.pipe_names)
model.to_disk("saved_model")
model = English().from_disk("saved_model")
print("Pipeline after loading from disk:", model.pipe_names)
```

The result is:
```
Original pipeline: ['tagger', 'parser', 'ner']
Pipeline after loading from disk: []
```

## Your Environment
* **spaCy version:** 2.3.0
* **Platform:** Linux-4.15.0-108-generic-x86_64-with-Ubuntu-18.04-bionic
* **Python version:** 3.6.9

**Comment by adrianeboyd** [comment, marked as answer]
When you initialize a model with `English()` it doesn't have any pipeline components, so it doesn't know to load them when you call `from_disk()`. If you add the right pipeline components and then run `from_disk()`, it will deserialize any that are in the current pipeline. See the example here: https://spacy.io/usage/saving-loading#pipeline

There can be cases where you want to handle things differently, but in general you probably want to use `spacy.load()` to load models, since it takes care of these details for you. The docs are here: https://spacy.io/api/top-level#spacy.load and you can see what it's doing underneath here:

https://github.com/explosion/spaCy/blob/2d715451a2215bf589da5e7d2c7a0234d05cbbc8/spacy/util.py#L194-L222
